src/FormTaxonomies/Actions/UpdateFormTaxonomies.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Give\FormTaxonomies\Actions;
+
+use Give\DonationForms\Models\DonationForm;
+
+/**
+ * @unreleased
+ */
+class UpdateFormTaxonomies
+{
+    /**
+     * @var array
+     */
+    protected $settings;
+
+    /**
+     * @unreleased
+     */
+    public function __construct(array $settings)
+    {
+        $this->settings = $settings;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function __invoke(DonationForm $form)
+    {
+        if(isset($this->settings['formTags'])) {
+            wp_set_post_terms($form->id, array_column($this->settings['formTags'], 'id'), 'give_forms_tag');
+        }
+
+        if(isset($this->settings['formCategories'])) {
```

Comment:
The IDs are already validated for hierarchical terms, but not flat terms.

The sure this up, I added a shared validation (since both are expected arrays of IDs) and bypassed `wp_set_post_terms` to call `wp_set_object_terms` directly with the validated terms.

See [25b5b0b](https://github.com/impress-org/givewp/pull/7463/commits/25b5b0b185546a2cb02681b5145a8f36fd247f0d).